### PR TITLE
CE: add docs for configuration auto-detection and setDataDir

### PR DESCRIPTION
**Pull request by heruan**
Update CE documentation reflecting the latest changes in configuration.

### CODE REVIEW

File: articles/ce/going-to-production.asciidoc [hunk 1]
```diff
@@ -53,26 +53,56 @@ You can set the path as a system property in a `VaadinServiceInitListener`, whic
...
 
-If you have a Spring application, you can register a `VaadinServiceInitListener` by implementing the interface and annotating the class as a `@SpringComponent`.
-In the `serviceInit` method, you can then set `vaadin.ce.dataDir` as a system property.
+If you have a Spring application, you can provide a bean of type `CollaborationEngineConfiguration` from your main Spring class
+by annotating a method with the `@Bean` annotation and returning a configuration instance with the data directory path set using the `setDataDir` method.
 
-.`com.company.myapp.MyVaadinInitListener.java`
-```java
-@SpringComponent
+.com.company.myapp.Application.java
```
Comment (jouni) [inline articles/ce/going-to-production.asciidoc]:
```suggestion
.`com.company.myapp.Application.java`
```

File: articles/ce/going-to-production.asciidoc [hunk 2]
```diff
@@ -53,26 +53,56 @@ You can set the path as a system property in a `VaadinServiceInitListener`, whic
...
+    }
+}
+----
+
+==== Other Applications
+
+If you have a non-Spring application, you can implement a `VaadinServiceInitListener`
+where you create a `CollaborationEngineConfiguration` instance and set it using the `CollaborationEngine.configure()` method.
+
+.com.company.myapp.MyVaadinInitListener.java
```
Comment (jouni) [inline articles/ce/going-to-production.asciidoc]:
```suggestion
.`com.company.myapp.MyVaadinInitListener.java`
```

File: articles/ce/going-to-production.asciidoc [hunk 3]
```diff
@@ -53,26 +53,56 @@ You can set the path as a system property in a `VaadinServiceInitListener`, whic
...
+[source,java,subs="+macros"]
```
Comment (jouni) [inline articles/ce/going-to-production.asciidoc:87]:
```suggestion
[source, Java, subs="+macros"]
```

File: articles/ce/going-to-production.asciidoc [hunk 4]
```diff
@@ -53,26 +53,56 @@ You can set the path as a system property in a `VaadinServiceInitListener`, whic
...
-If you have a Spring application, you can register a `VaadinServiceInitListener` by implementing the interface and annotating the class as a `@SpringComponent`.
-In the `serviceInit` method, you can then set `vaadin.ce.dataDir` as a system property.
+If you have a Spring application, you can provide a bean of type `CollaborationEngineConfiguration` from your main Spring class
+by annotating a method with the `@Bean` annotation and returning a configuration instance with the data directory path set using the `setDataDir` method.
 
-.`com.company.myapp.MyVaadinInitListener.java`
-```java
-@SpringComponent
+.com.company.myapp.Application.java
+[source,java,subs="+macros"]
```
Comment (jouni) [inline articles/ce/going-to-production.asciidoc:60]:
```suggestion
[source, Java, subs="+macros"]
```

File: articles/ce/going-to-production.asciidoc [hunk 5]
```diff
@@ -181,25 +211,12 @@ One potential way to handle the event is to send a message to any existing appli
...
 
-The listener can be configured in a `VaadinServiceInitListener` in the same way as the `vaadin.ce.dataDir` property,
-if you're setting that property in Java code, as described earlier.
-
-The following example is a Spring project, so the `VaadinServiceInitListener` is registered by adding the `@SpringComponent` annotation.
-If you are not using Spring, you can register the service init listener in the same way as described in <<ce.production.data-dir-in-project-files>>.
+The listener can be configured when creating the `CollaborationEngineConfiguration` as described earlier
+in <<ce.production.data-dir-in-project-files>>.
 
 ```java
```
Comment (jouni) [inline articles/ce/going-to-production.asciidoc:217]:
```suggestion
[source, Java]
----
```

File: articles/ce/going-to-production.asciidoc [hunk 6]
```diff
@@ -213,17 +230,7 @@ public static class MyVaadinInitListener
...
-                licenseEventHandler);
-        CollaborationEngine.configure(service, configuration);
-    }
-
-    private void sendEmail(String subject, String content) {
-        // Implement sending an email to relevant people
-    }
-}
+    });
 ```
```
Comment (jouni) [inline articles/ce/going-to-production.asciidoc:234]:
```suggestion
----
```

**Review by jouni — APPROVED** [review]

**Comment by pekam** [comment]
These features are not released yet. I pushed a new branch `ce-next` that is up to date with `master`. Let's change the target branch to that one.

Also, the commit/PR title should tell the features that are documented here. E.g. "CE: add docs for configuration auto-detection and setDataDir".
Anyway, I suppose we'll squash this, so the final commit message can be edited at that point.

**Review by pekam — APPROVED** [review]

**Pull request merged**